compiler/rustc_type_ir/src/search_graph/global_cache.rs

```diff
@@ -53,14 +53,21 @@ impl<X: Cx> GlobalCache<X> {
     ) {
         let result = cx.mk_tracked(result, dep_node);
         let entry = self.map.entry(input).or_default();
+        let result_tracked = cx.get_tracked(&result);
```

Comment:
use the original input result :sweat_smile: maybe call it "raw_result" or whatever